cpu/numactl.py

```diff
@@ -75,3 +131,94 @@ def test(self):
+        '''
+        cmd = "numactl --interleave=all ping -I %s %s -c %s -f"\
+              % (self.iface, self.peer, self.ping_count)
+        self.numa_ping(cmd)
+
+    def test_localalloc(self):
+        '''
+        Test memory allocation on the current node
+        '''
+        cmd = "numactl --localalloc ping -I %s %s -c %s -f"\
+              % (self.iface, self.peer, self.ping_count)
+        self.numa_ping(cmd)
+
+    def test_preferred_node(self):
+        '''
+        Test Preferably allocate memory on node
+        '''
+        if self.check_numa_nodes():
+            self.node_number = [key for key in self.numa_dict.keys()][1]
+            cmd = "numactl --preferred=%s  ping -I %s %s -c %s -f" \
+                % (self.node_number, self.iface, self.peer, self.ping_count)
+            self.numa_ping(cmd)
+
+    def test_cpunode_with_membind(self):
+        '''
+        Test CPU and memory bind
+        '''
+        if self.check_numa_nodes():
+            self.first_cpu_node_number = [key
+                                          for key
+                                          in self.numa_dict.keys()][0]
+            self.second_cpu_node_number = [key
+                                           for key
+                                           in self.numa_dict.keys()][1]
+            self.membind_node_number = [key
+                                        for key
+                                        in self.numa_dict.keys()][1]
+            for cpu in [self.first_cpu_node_number,
+                        self.second_cpu_node_number]:
+                cmd = "numactl --cpunodebind=%s --membind=%s "\
+                      "ping -I %s %s -c %s -f" % (cpu,
+                                                  self.membind_node_number,
+                                                  self.iface,
+                                                  self.peer,
+                                                  self.ping_count)
+                self.numa_ping(cmd)
+
+    def test_physical_cpu_bind(self):
+        '''
+        Test physcial  CPU binds
+        '''
+        if self.check_numa_nodes():
+            self.cpu_number = [value
+                               for value
+                               in self.numa_dict.values()][0][1]
+            cmd = "numactl --physcpubind=%s ping -I %s %s -c %s -f"\
+                % (self.cpu_number, self.iface, self.peer, self.ping_count)
+            self.numa_ping(cmd)
+
+    def test_numa_pci_bind(self):
+        '''
+        Test PCI binding to diferrent NUMA nodes
+        '''
+        if self.check_numa_nodes():
+            nodes = [node for node in self.numa_dict.keys()]
+            pci_node_number = genio.read_file(
+                                              "/sys/bus/pci/devices/%s/"
+                                              "numa_node" % self.device)
+            node_path = '/sys/bus/pci/devices/%s/numa_node' % self.device
```

Comment:
node_path can go one line up and than simplify to
node_path = '/sys/bus/pci/devices/%s/numa_node' % self.device
pci_node_number = genio.read_file(node_path)

Reply:
@abdhaleegit fixed